diff --git a/src/Options.d.ts b/src/Options.d.ts
index abfb7a23b94e2244d3529eb92ce540b7778f27c2..ca546fcc2b54d447a32f4c51abccdbbdddea55a2 100644
--- a/src/Options.d.ts
+++ b/src/Options.d.ts
@@ -1,4 +1,6 @@
-import { ZodSchema } from "zod";
+import { ZodSchema, type ZodTypeDef } from "zod";
+import type { Refs } from "./Refs";
+import type { JsonSchema7Type } from "./parseDef";
 export type Targets = "jsonSchema7" | "jsonSchema2019-09" | "openApi3";
 export type Options<Target extends Targets = "jsonSchema7"> = {
     name: string | undefined;
@@ -14,6 +16,7 @@ export type Options<Target extends Targets = "jsonSchema7"> = {
     errorMessages: boolean;
     markdownDescription: boolean;
     emailStrategy: "format:email" | "format:idn-email" | "pattern:zod";
+    onParseDef: ((def: ZodTypeDef, refs: Refs, schema: JsonSchema7Type) => void) | undefined;
 };
 export declare const defaultOptions: Options;
 export declare const getDefaultOptions: <Target extends Targets>(options: string | Partial<Options<Target>> | undefined) => Options<Target>;
diff --git a/src/Options.js b/src/Options.js
index 8b680c494e08a7b068cd505eef9c37a19be40523..ed84e25390fbd99f0c0f8f233dd43f219b8e1234 100644
--- a/src/Options.js
+++ b/src/Options.js
@@ -15,6 +15,7 @@ exports.defaultOptions = {
     errorMessages: false,
     markdownDescription: false,
     emailStrategy: "format:email",
+    onParseDef: undefined,
 };
 const getDefaultOptions = (options) => (typeof options === "string"
     ? Object.assign(Object.assign({}, exports.defaultOptions), { name: options }) : Object.assign(Object.assign({}, exports.defaultOptions), options));
diff --git a/src/parseDef.js b/src/parseDef.js
index 4f33785322590999d31593cdcbdb174175930799..26748a81097c8264835ea3e059bc8440fae00d54 100644
--- a/src/parseDef.js
+++ b/src/parseDef.js
@@ -45,6 +45,7 @@ function parseDef(def, refs, forceResolution = false // Forces a new schema to b
     const jsonSchema = selectParser(def, def.typeName, refs);
     if (jsonSchema) {
         addMeta(def, refs, jsonSchema);
+        if (refs.onParseDef) refs.onParseDef(def, refs, jsonSchema);
     }
     newItem.jsonSchema = jsonSchema;
     return jsonSchema;